daemon/api.go

```diff
@@ -157,6 +164,46 @@ func sysInfo(c *Command, r *http.Request) Response {
 	return SyncResponse(m)
 }
 
+func loginUser(c *Command, r *http.Request) Response {
+	var loginData struct {
+		Username string `json:"username"`
+		Password string `json:"password"`
+		Otp      string `json:"otp"`
+	}
+
+	decoder := json.NewDecoder(r.Body)
+	if err := decoder.Decode(&loginData); err != nil {
+		return BadRequest("can't decode login data from request body: %v", err)
+	}
+
+	macaroon, err := store.RequestPackageAccessMacaroon()
+	if err != nil {
+		return InternalError("couldn't get package access macaroon")
+	}
+
+	discharge, err := store.DischargeAuthCaveat(loginData.Username, loginData.Password, macaroon, loginData.Otp)
+	if err == store.ErrAuthenticationNeeds2fa {
+		return Unauthorized("TWOFACTOR_REQUIRED")
+	}
+	if err != nil {
+		return Unauthorized("couldn't get discharge authorization")
+	}
+
+	discharges := []string{discharge}
+	macaroons := map[string]interface{}{
```

Comment:
Let's please define a private global struct instead of a map:
```
type macaroonData struct {
        Macaroon string `json:"macaroon"`
        Discharges []string `json:"discharges"`
}
```